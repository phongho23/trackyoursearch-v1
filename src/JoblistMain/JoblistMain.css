
.JoblistMain ul {
  list-style-type: none;
  margin: 0;
  padding: 10px;
}

.JoblistMain__button-container {
  max-width: 380px;
  margin: 0 10px;
}

.JoblistMain__add-job-button {
  margin: 0 auto 20px;
}
.JoblistMain__add-job-button:hover {
  background-color:purple;
  color: white;
}
