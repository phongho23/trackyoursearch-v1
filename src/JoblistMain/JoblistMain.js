//watch out for 'job' items below

import React from 'react'
import { Link } from 'react-router-dom'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import Jobitem from '../Jobitem/Jobitem'
import CircleButton from '../CircleButton/CircleButton'
import './JoblistMain.css'

export default function JoblistMain(props) {
  return (
    <section className='JoblistMain'>
      <div className='JoblistMain__button-container'>
        <CircleButton
          tag={Link}
          to='/home/add-job'
          type='button'
          className='JoblistMain__add-job-button'>
          <FontAwesomeIcon icon='plus' />
          <br />
          Job
        </CircleButton>
      </div>
      <ul style={{display: 'flex', flexWrap: 'wrap'}}>
        {props.jobopps.map(job =>
          <li style={{display: "inline-block"}} key={job.id}>
            <Jobitem
              id={job.id}
              name={job.name}
              modified={job.modified}
            />
          </li>
        )}
      </ul>
    </section>
  )
}

JoblistMain.defaultProps = {
  jobopps: [],
}
