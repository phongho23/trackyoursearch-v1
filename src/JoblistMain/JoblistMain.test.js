import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import JoblistMain from './JoblistMain'

describe(`JoblistMain component`, () => {
  const props = {
    jobopps: [
      {
        "id": "cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1",
        "name": "Dogs",
        "modified": "2019-01-03T00:00:00.000Z",
        "weekId": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
        "content": "Corporis accusamus placeat.\n \rUnde."
      },
      {
        "id": "d26e0034-ffaf-11e8-8eb2-f2801f1b9fd1",
        "name": "Cats",
        "modified": "2018-08-15T23:00:00.000Z",
        "weekId": "b07161a6-ffaf-11e8-8eb2-f2801f1b9fd1",
        "content": "Eos\n \rlaudantium."
      },
      {
        "id": "d26e01a6-ffaf-11e8-8eb2-f2801f1b9fd1",
        "name": "Pigs",
        "modified": "2018-03-01T00:00:00.000Z",
        "weekId": "b07161a6-ffaf-11e8-8eb2-f2801f1b9fd1",
        "content": "Occaecati dignissimos\nvoluptatum nihil."
      },
      {
        "id": "d26e0570-ffaf-11e8-8eb2-f2801f1b9fd1",
        "name": "Birds",
        "modified": "2019-01-04T00:00:00.000Z",
        "weekId": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
        "content": "Eum culpa odit."
      },
    ]
  }

  it('renders a .JoblistMain by default', () => {
    const wrapper = shallow(<JoblistMain />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders a Jobitem in ul for each job in array', () => {
    const ul = shallow(<JoblistMain {...props} />)
      .find('ul')
    expect(toJson(ul)).toMatchSnapshot()
  })
})
