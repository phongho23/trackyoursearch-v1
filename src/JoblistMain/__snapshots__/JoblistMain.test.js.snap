// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JoblistMain component renders a .JoblistMain by default 1`] = `
<section
  className="JoblistMain"
>
  <ul />
  <div
    className="JoblistMain__button-container"
  >
    <NavCircleButton
      className="JoblistMain__add-job-button"
      tag={[Function]}
      to="/add-job"
      type="button"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="plus"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        title=""
        transform={null}
      />
      <br />
      Job
    </NavCircleButton>
  </div>
</section>
`;

exports[`JoblistMain component renders a Job in ul for each jobopps in array 1`] = `
<ul>
  <li
    key="cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1"
  >
    <Jobitem
      id="cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1"
      modified="2019-01-03T00:00:00.000Z"
      name="Dogs"
    />
  </li>
  <li
    key="d26e0034-ffaf-11e8-8eb2-f2801f1b9fd1"
  >
    <Jobitem
      id="d26e0034-ffaf-11e8-8eb2-f2801f1b9fd1"
      modified="2018-08-15T23:00:00.000Z"
      name="Cats"
    />
  </li>
  <li
    key="d26e01a6-ffaf-11e8-8eb2-f2801f1b9fd1"
  >
    <Jobitem
      id="d26e01a6-ffaf-11e8-8eb2-f2801f1b9fd1"
      modified="2018-03-01T00:00:00.000Z"
      name="Pigs"
    />
  </li>
  <li
    key="d26e0570-ffaf-11e8-8eb2-f2801f1b9fd1"
  >
    <Jobitem
      id="d26e0570-ffaf-11e8-8eb2-f2801f1b9fd1"
      modified="2019-01-04T00:00:00.000Z"
      name="Birds"
    />
  </li>
</ul>
`;
